src/TLuaInterpreter.cpp

```diff
@@ -9897,41 +9973,67 @@ int TLuaInterpreter::getAllMapUserData( lua_State * L )
 
 int TLuaInterpreter::downloadFile( lua_State * L )
 {
-    string path, url;
-    if( ! lua_isstring( L, 1 ) )
-    {
-        lua_pushstring( L, "downloadFile: wrong argument type" );
+    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
+    if( ! pHost ) {
+        lua_pushnil( L );
+        lua_pushstring( L, tr( "downloadFile: NULL Host pointer - something is wrong!" )
+                        .toUtf8().constData() );
+        return 2;
+    }
+
+    QString localFile;
+    if( ! lua_isstring( L, 1 ) ) {
+        lua_pushstring( L, tr( "downloadFile: bad argument #1 type (local filename as string expected, got %1!)" )
+                        .arg( luaL_typename( L, 1 ) )
+                        .toUtf8().constData() );
         lua_error( L );
         return 1;
     }
-    else
-    {
-        path = lua_tostring( L, 1 );
+    else {
+        localFile = QString::fromUtf8( lua_tostring( L, 1 ) );
     }
-    if( ! lua_isstring( L, 2 ) )
-    {
-        lua_pushstring( L, "downloadFile: wrong argument type" );
+
+    QString urlString;
+    if( ! lua_isstring( L, 2 ) ) {
+        lua_pushstring( L, tr( "downloadFile: bad argument #1 type (remote url as string expected, got %1!)" )
+                        .arg( luaL_typename( L, 2 ) )
+                        .toUtf8().constData() );
         lua_error( L );
         return 1;
     }
-    else
-    {
-        url = lua_tostring( L, 2 );
+    else {
+        urlString = QString::fromUtf8( lua_tostring( L, 2 ) );
     }
-    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
-    QString _url = url.c_str();
-    QString _path = path.c_str();
-    QNetworkRequest request = QNetworkRequest( QUrl( _url ) );
+
+    QUrl url = QUrl::fromUserInput( urlString );
+
+    if( ! url.isValid() ) {
```

Comment:
Since there is the facility to test for a valid `url` it seemed silly not to use it, IMHO!